libsolidity/analysis/TypeChecker.cpp

```diff
@@ -343,18 +343,56 @@ void TypeChecker::annotateBaseConstructorArguments(
 	{
 		ASTNode const* previousNode = insertionResult.first->second;
 
+		bool isInCurrentContract = false;
```

Comment:
This is rather complicated. A slightly more hackish solution could use `SourceLocation::contains`.